### PR TITLE
node v20.16.0

### PR DESCRIPTION
**Pull request by DJoyekurun**
- Upgrade node version
- Remove circle ci pipeline (Because it takes as much time as github actions to create multi-arch images)
- Fix warnings

https://github.com/SharinPix/SharinPix/pull/12782

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -46,9 +46,9 @@ RUN bash -c "curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rb
 
 # Node
 ENV NVM_DIR /home/user/.nvm
```
Comment (DJoyekurun) [inline Dockerfile]: image

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,73 +1,48 @@
 version: 2.1
+
 executors:
-  docker-publisher:
-    environment:
-      IMAGE_NAME: sharinpix/docker-heroku
+  docker-executor:
     docker:
-      - image: circleci/buildpack-deps:stretch
```
Comment (DJoyekurun) [inline .circleci/config.yml:7]: Legacy images with the prefix “circleci/” were [deprecated](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034)
 on December 31, 2021.

File: Dockerfile [hunk 3]
```diff
@@ -67,4 +67,4 @@ WORKDIR /app
 
 EXPOSE 5000
 
-CMD bash
```
Comment (DJoyekurun) [inline Dockerfile:70]: https://docs.docker.com/reference/build-checks/json-args-recommended/

File: Dockerfile [hunk 4]
```diff
@@ -45,10 +45,10 @@ RUN bash -c "curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rb
   bash -c "/home/user/.rbenv/shims/gem install bundler"
 
 # Node
-ENV NVM_DIR /home/user/.nvm
```
Comment (DJoyekurun) [inline Dockerfile:48]: https://docs.docker.com/reference/build-checks/legacy-key-value-format/

**Review by zfir — APPROVED** [review]

File: .github/workflows/docker-image.yml [hunk 5]
```diff
@@ -16,9 +18,9 @@ jobs:
...
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       - name: Build and push
-        uses: docker/build-push-action@v5
+        uses: docker/build-push-action@v6
         with:
           context: .
           platforms: linux/amd64,linux/arm64
           push: true
-          tags: sharinpix/docker-heroku:pr-gh-${{ github.event.number }}
+          tags: sharinpix/docker-heroku:${{ (github.ref == 'refs/heads/main') && 'latest' || format('pr-gh-{0}', github.run_number) }}
```
Comment (DJoyekurun) [inline .github/workflows/docker-image.yml:26]: this is github workflows' way of doing ternary conditions

**Review by ombr — APPROVED** [review]

**Review by keshav1720 — APPROVED** [review]

**Review by ashishven — APPROVED** [review]

**Review by atishdhu — APPROVED** [review]

**Pull request merged**